tests/Admin/AdminTest.php

```diff
@@ -1294,35 +1294,56 @@ public function testDeterminedPerPageValue(): void
     public function testIsGranted(): void
     {
         $admin = new PostAdmin('sonata.post.admin.post', 'Acme\NewsBundle\Entity\Post', 'Sonata\NewsBundle\Controller\PostAdminController');
+        $modelManager = $this->createMock(ModelManagerInterface::class);
+        $modelManager
+            ->method('getNormalizedIdentifier')
+            ->willReturnCallback(static function (?object $entity = null): ?string {
+                return $entity ? $entity->id : null;
+            });
 
-        $entity = new \stdClass();
+        $admin->setModelManager($modelManager);
 
-        $securityHandler = $this->createMock(AclSecurityHandlerInterface::class);
+        $entity1 = new \stdClass();
+        $entity1->id = '1';
+
+        $securityHandler = $this->createStub(AclSecurityHandlerInterface::class);
```

Comment:
```suggestion
        $securityHandler = $this->createMock(AclSecurityHandlerInterface::class);
```

Reply:
Since this test double performs expectations about its methods, I prefer to leverage `createStub()` in order to be aligned with the pattern mentioned at sebastianbergmann/phpunit#3120.

Reply:
That's why I suggested leave it as `createMock` since it has an `expects` call.

Reply:
When you're right ... :laughing: 
The call is up to date. Thank you!